core/src/apps/management/common/layout.py

```diff
@@ -290,14 +352,80 @@ def _get_mnemonic_page(words: list):
     return count
 
 
-async def slip39_prompt_threshold(ctx, num_of_shares):
+async def slip39_prompt_number_of_groups(ctx):
+    count = 5
+    min_count = 2
+    max_count = 16
+
+    while True:
+        shares = ShamirNumInput(ShamirNumInput.SET_GROUPS, count, min_count, max_count)
+        confirmed = await confirm(
+            ctx,
+            shares,
+            ButtonRequestType.ResetDevice,
+            cancel="Info",
+            confirm="Continue",
+            major_confirm=True,
+            cancel_style=ButtonDefault,
+        )
+        count = shares.input.count
+        if confirmed:
+            break
+
+        info = InfoConfirm(
+            "Group contains set "
+            "number of shares and "
+            "its own threshold. "
+            "In the next step you set "
+            "both number of shares "
+            "and threshold."
+        )
+        await info
+
+    return count
+
+
+async def slip39_prompt_group_threshold(ctx, num_of_groups):
+    count = num_of_groups // 2 + 1
+    min_count = 2
+    max_count = num_of_groups
+
+    while True:
+        shares = ShamirNumInput(
+            ShamirNumInput.SET_GROUP_THRESHOLD, count, min_count, max_count
+        )
+        confirmed = await confirm(
+            ctx,
+            shares,
+            ButtonRequestType.ResetDevice,
+            cancel="Info",
+            confirm="Continue",
+            major_confirm=True,
+            cancel_style=ButtonDefault,
+        )
+        count = shares.input.count
+        if confirmed:
+            break
+        else:
+            info = InfoConfirm(
+                "Group threshold "
+                "specifies number of "
+                "groups required "
+                "to recover wallet. "
+            )
+            await info
+
+    return count
+
+
+async def slip39_prompt_threshold(ctx, num_of_shares, group_id=None):
     count = num_of_shares // 2 + 1
     min_count = 2
```

Comment:
This should be something like: min_count = min(2, num_of_shares)
In fact, if num_of_shares is 1, then we don't even have to prompt for the threshold. The same applies if num_of_shares is 2, but in that case it's better for the user to explicitly confirm threshold = 2.

Reply:
this function is only used in case of single group shamir so I actually haven't touched it and it's been this way in shamir beta. If it's not a critical issue we should resolve it as part of #442, I'll add it to the list.